elyra/pipeline/pipeline.py

```diff
@@ -99,6 +100,8 @@ def classifier(self):
 
     @property
     def name(self):
+        if self._name == os.path.basename(self._filename):
+            return os.path.basename(self._name).split(".")[0]
```

Comment:
I suggest setting `_name` here, otherwise this calculation will happen on each property access.
```suggestion
            self._name = os.path.basename(self._name).split(".")[0]
```

Reply:
fixed